modules/ROOT/pages/custom-library-development.adoc

```diff
@@ -0,0 +1,730 @@
+:description: Develop a custom library project that will be used in a Bonita project and mutualize advanced business rules capabilities.
+
+{description}
+
+== Why should I develop a custom library ?
+
+* To extract advanced logic from processes script expressions
+* To be able to use standard technology to automate testing for this logic
+* To be able to reuse and share this logic between several processes and Bonita projects.
+* To better manage my library dependencies
+
+== Initialize the project
+
+[NOTE]
+====
+In the below example we use Maven as the project build tool, but it is also possible Gradle if it is the tool of your choice.
+====
+ 
+The example will externalize a function used in the xref:define-who-can-do-what.adoc[Define who can do what] article.
+ 
+=== Choose your preferred tools
+
+You are free to choose the IDE of your choice to develop your library (Eclipse IDE, IntelliJ IDEA, VS Code...).
+You can also choose between `Java 11` or `Groovy 3` as target programming language.
+The following example will use `Java` and `Maven`, but other options are possible. 
+
+To follow this tutorial you will need to install:
+
+* A https://adoptopenjdk.net/[JDK 11]
+* https://maven.apache.org/download.cgi[Maven] 
+* The https://git-scm.com/downloads[Git SCM Client] (Optional)
+
+An Internet connection with access to Maven Central is required.
+
+=== Basic setup
+
+Create a new folder on your file system, and call it `my-first-bonita-lib`.
+At the root of this folder create a `pom.xml` file.
+
+[source, xml]
+.pom.xml
+----
+<?xml version="1.0" encoding="UTF-8"?>
+
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+
+    <groupId>org.bonitasoft.extension</groupId> <1>
+    <artifactId>bonita-extension-library</artifactId> <1>
+    <version>1.0.0-SNAPSHOT</version>
+
+    <name>Bonita extension function library</name>
+
+    <properties>
+        <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+        <project.reporting.outputEncoding>UTF-8</project.reporting.outputEncoding>
+        <java.version>11</java.version>
+        <maven.compiler.source>${java.version}</maven.compiler.source>
+        <maven.compiler.target>${java.version}</maven.compiler.target>
+    </properties>
+
+</project>
+----
+<1> You may choose your own `groupId` and `artifactId`
+
+== Integration with Bonita
+
+=== Using the Runtime BOM
+
+This library will be executed in the Bonita Runtime that already comes with `provided` dependencies in its class-path.
+To avoid dependencies conflicts between our custom library and the Bonita Runtime, we can use a **Bill Of Material** (BOM).
+
+First, we need to define the target Bonita runtime version. Add a `bonita-runtime.version` property in `properties` section of the pom.xml:
+
+[source, xml]
+----
+<properties>
```

Comment:
This section should point out to a table for the user to master engine version versus platform version.